editoast/authz/src/regulator.rs

```diff
@@ -103,6 +110,14 @@ pub trait StorageDriver: Clone {
     -> impl Future<Output = Result<bool, Self::Error>> + Send;
 }
 
+/// Check if an entity (user, group, or infra) is orphaned.
+fn is_entity_orphaned(entity_str: &str, prefix: &str, existing_ids: &HashSet<i64>) -> bool {
+    entity_str
+        .strip_prefix(prefix)
+        .and_then(|s| s.split('#').next()?.parse().ok())
+        .is_some_and(|id| !existing_ids.contains(&id))
+}
```

Comment:
Idealy we could have a function that deserialize the `entity_str` so we can have the type and the id, and then check if it's part of the existing ids for a given entity type.

@leovalais maybe you have an idea on how do this?